integrations/unstructured/src/haystack_integrations/components/converters/unstructured/converter.py

```diff
@@ -107,19 +117,20 @@ def run(self, paths: Union[List[str], List[os.PathLike]]):
 
         # currently, the files are converted sequentially to gently handle API failures
         documents = []
+        meta_list = normalize_metadata(meta, sources_count=len(all_filepaths))
```

Comment:
If `len(all_filepaths) != len meta`, `normalize_metadata` method fails with `ValueError`.
This is expected and right for common converters, but this component also loops all files in the provided directories, so this case would be quite common and I would not want the converter to fail.

I would propose a slightly different approach. Something similar to this (untested):

```python
try:
    meta_list = normalize_metadata(meta, sources_count=len(all_filepaths))
except ValueError as e:
    logger.warning("The length of the metadata list does not match the length of all filepaths (also files inside directories are included in all filepaths). The metadata list will be ignored. ")
    meta_list = [{}] * all_filepaths
```

Reply:
> If `len(all_filepaths) != len meta`, `normalize_metadata` method fails with `ValueError`. This is expected and right for common converters, but this component also loops all files in the provided directories, so this case would be quite common and I would not want the converter to fail.
> 
> I would propose a slightly different approach. Something similar to this (untested):
> 
> ```python
> meta_list = [{}] * all_filepaths
> try:
>     meta_list = normalize_metadata(meta, sources_count=len(all_filepaths))
> except ValueError as e:
>     logger.warning("The length of the metadata list does not match the length of all filepaths (also files inside directories are included in all filepaths). The metadata list will be ignored. ")
> ```

I've added a test where the path is a directory.
Related to metadata handling hm, I disagree a little bit.
How it works currently (with my merge request): if there are no meta provided, it works without error and doesn't add custom metadata to documents. If metadata are provided by the user, they should match the number of files given or the number of files in the directory path, otherwiser it raises an error.

As a user of Haystack, I would much rather have a component crash and tell me "you're making a mistake" than silently working, but not the way I expected. Realising after indexing files that now my document store is full of documents without metadata because the component decided to handle missing metadata by ignoring them sounds painful to me. Because then I would have to clean my docstore and re-index my files: but cleaning part of the docstore when documents doesn't have metadata is... hard). 

Also I feel like if it only throw a Warning and doesn't raise error, I would only realise the issue with my indexation way later, through a query pipeline throwing weird results or metadata filtering not working as expected.

Reply:
Thinking about it more, I agree with you...
Let's just improve the docstring to make the behavior more explicit.